src/test/java/org/prebid/server/settings/service/HttpPeriodicRefreshServiceTest.java

```diff
@@ -0,0 +1,184 @@
+        updatedResponse = HttpClientResponse.of(200, null,
+                mapper.writeValueAsString(HttpRefreshResponse.of(
+                        singletonMap("id1", mapper.createObjectNode().put("deleted", "true")),
+                        singletonMap("id2", mapper.createObjectNode().put("field2", "field-value2")))));
+
+        expectedRequests = singletonMap("id1", "{\"field1\":\"field-value1\"}");
+        expectedImps = singletonMap("id2", "{\"field2\":\"field-value2\"}");
+
+        given(httpClient.get(eq(ENDPOINT_URL), anyLong()))
+                .willReturn(Future.succeededFuture(httpRefreshResponse));
+        given(httpClient.get(contains("?last-modified="), anyLong()))
+                .willReturn(Future.succeededFuture(updatedResponse));
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() -> createAndInitService(cacheNotificationListener,
+                "invalid_url", 1, 1, vertx, httpClient));
+    }
+
+    @Test
+    public void shouldCallSaveWithExpectedParameters() {
+        // when
+        httpPeriodicRefreshService = createAndInitService(cacheNotificationListener, ENDPOINT_URL,
+                1000, 2000, vertx, httpClient);
+
+        // then
+        verify(cacheNotificationListener).save(expectedRequests, expectedImps);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void shouldCallInvalidateAndSaveWithExpectedParameters() {
+        // given
+        given(vertx.setPeriodic(anyLong(), any()))
+                .willAnswer(inv -> {
+                    ((Handler<Long>) inv.getArgument(1)).handle(0L);
+                    return 0L;
+                });
+
+        // when
+        httpPeriodicRefreshService = createAndInitService(cacheNotificationListener, ENDPOINT_URL,
+                1000, 2000, vertx, httpClient);
+
+        // then
+        verify(cacheNotificationListener).save(expectedRequests, expectedImps);
+        verify(cacheNotificationListener).invalidate(singletonList("id1"), emptyList());
+        verify(cacheNotificationListener).save(emptyMap(), expectedImps);
+
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void shouldCallSaveAfterUpdate() throws JsonProcessingException {
+        // given
+        updatedResponse = HttpClientResponse.of(200, null,
+                mapper.writeValueAsString(HttpRefreshResponse.of(
+                        singletonMap("id1", mapper.createObjectNode().put("changed1", "value-changed2")),
+                        singletonMap("id2", mapper.createObjectNode().put("field2", "field-value2")))));
+
+        given(vertx.setPeriodic(anyLong(), any()))
+                .willAnswer(inv -> {
+                    ((Handler<Long>) inv.getArgument(1)).handle(0L);
+                    return 0L;
+                });
+        given(httpClient.get(startsWith(ENDPOINT_URL + "?last-modified="), anyLong()))
```

Comment:
Here `startsWith(ENDPOINT_URL + "?last-modified=")` matcher is used while in `setUp` it is different - `contains("?last-modified=")`. If there is no good reason to have both variants please choose one and stick to it everywhere.